src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -62,6 +81,43 @@ void check_sshfs_exists(mp::SSHSession& session)
     run_cmd(session, "which sshfs", error_handler);
 }
 
+// If the target name starts with ~, this function returns the full directory name. If not, it returns the same
+// target name.
+std::string expand_home_directory(mp::SSHSession& session, const std::string& target)
+{
+    std::string expanded_target;
+
+    if ('~' == target[0])
+    {
+        std::string home;
+
+        // Get the user name, which can be empty.
+        auto pos = (1 == target.size()) ? 1 : target.find('/', 1);
```

Comment:
Maybe overkill, but use a regular expression here? Would make for more concise code on our part.

Something like [`^~([^/]+)?(.+)?$`](https://regex101.com/r/P2pk8f/1/tests) should do.

Reply:
Yes, it can be done more concisely and elegantly.

Reply:
Another thing: why not replacing all by `run_cmd(session, fmt::format("echo {}", target))`?

Reply:
`readlink` doesn't expand `~` itself, and if we go for `bash` expanding it, we don't need any echo, just remove the quotes from `P="{}"…` below.